src/eventHandlers/InterviewEventHandler.js

```diff
@@ -316,8 +486,21 @@ async function processRequest (payload) {
  * @returns {undefined}
  */
 async function processUpdate (payload) {
-  await sendInterviewScheduledNotifications(payload)
-  await checkOverlapping(payload)
+  const interview = payload.value
+  const interviewOldValue = payload.options.oldValue
+  if (_.includes([Constants.Interviews.Status.Scheduled, Constants.Interviews.Status.Rescheduled], interview.status) &&
+interview.status === interviewOldValue.status && !moment(interview.startTimestamp).isSame(interviewOldValue.startTimestamp)) {
+    // rescheduled
+    await sendInterviewRescheduledNotifications(payload)
+    await checkOverlapping(payload)
+  } else if (_.includes([Constants.Interviews.Status.Scheduled, Constants.Interviews.Status.Rescheduled], interviewOldValue.status) && interview.status === Constants.Interviews.Status.Cancelled) {
+    // cancelled
+    await sendInterviewCancelledNotifications(payload)
+  } else {
+    // scheduled
+    await sendInterviewScheduledNotifications(payload)
+    await checkOverlapping(payload)
+  }
```

Comment:
@yoution could we please keep all the logic inside methods, so there we just call methods like:
```js
async function processUpdate (payload) {
  await sendInterviewScheduledNotifications(payload)
  await sendInterviewRescheduledNotifications(payload)
  await sendInterviewCancelledNotifications(payload)
  await checkOverlapping(payload)
} 
```

And inside each new method `sendInterviewRescheduledNotifications` and `sendInterviewCancelledNotifications`, we must check if we need to send notification or not. Similar to how we do it inside method `sendInterviewScheduledNotifications`:

![image](https://user-images.githubusercontent.com/146016/141962744-a2c58f6a-ddd0-40ab-8a8f-67bc782c09c1.png)